core/src/verifier.rs

```diff
@@ -1165,4 +1191,121 @@ impl Owner for Verifier {
+
+            tracing::error!("Collecting deposits and withdrawals");
+            let new_deposits = citrea_client
+                .collect_deposit_move_txids(l2_height_start + 1, l2_height_end)
+                .await?;
+            tracing::error!("New Deposits: {:?}", new_deposits);
+            let new_withdrawals = citrea_client
+                .collect_withdrawal_utxos(l2_height_start + 1, l2_height_end)
+                .await?;
+            tracing::error!("New Withdrawals: {:?}", new_withdrawals);
+            for (idx, move_to_vault_txid) in new_deposits {
+                tracing::error!("Setting move to vault txid: {:?}", move_to_vault_txid);
+                self.db
+                    .set_move_to_vault_txid_from_citrea_deposit(
+                        Some(&mut dbtx),
+                        idx as u32,
+                        &move_to_vault_txid,
+                    )
+                    .await?;
+            }
+            for (idx, withdrawal_utxo_outpoint) in new_withdrawals {
+                tracing::error!("Setting withdrawal utxo: {:?}", withdrawal_utxo_outpoint);
+                self.db
+                    .set_withdrawal_utxo_from_citrea_withdrawal(
+                        Some(&mut dbtx),
+                        idx as u32,
+                        withdrawal_utxo_outpoint,
+                        block_height,
+                    )
+                    .await?;
+            }
+
+            tracing::error!("Getting payout txids");
+            let payout_txids = self
+                .db
+                .get_payout_txs_from_citrea_withdrawal(Some(&mut dbtx), block_id)
+                .await?;
+
+            let txid_to_tx_hashmap: HashMap<bitcoin::Txid, &bitcoin::Transaction> = block
+                .txdata
+                .iter()
+                .map(|tx| (tx.compute_txid(), tx))
+                .collect();
+
+            let mut payout_txs_and_payer_operator_idx = Vec::new();
+            for (idx, payout_txid) in payout_txids {
+                let payout_tx = txid_to_tx_hashmap
+                    .get(&payout_txid)
+                    .ok_or(BridgeError::Error("Payout tx not found".to_string()))?;
+                let last_output = &payout_tx.output[payout_tx.output.len() - 1]
+                    .script_pubkey
+                    .to_bytes();
+                tracing::error!("last_output: {}, idx: {}", hex::encode(last_output), idx);
+
+                let operator_idx = u32::from_le_bytes(
+                    last_output[2..6]
+                        .try_into()
+                        .expect("Failed to convert last_output to u32"),
+                );
+
+                payout_txs_and_payer_operator_idx.push((idx, payout_txid, operator_idx));
+            }
+
+            self.db
+                .set_payout_txs_and_payer_operator_idx(
+                    Some(&mut dbtx),
+                    payout_txs_and_payer_operator_idx,
+                )
+                .await?;
```

Comment:
nit: This function is a little long, I'd prefer the inner logic to be split. I believe we can have

```rs
async fn get_citrea_l2_height_range(citrea_client, l1_block_height) -> (u32, u32);
async fn update_citrea_withdrawals(citrea, l2_height_range);
async fn update_finalized_payouts(...);
```